springfox-swagger2/src/main/java/springfox/documentation/swagger2/web/Swagger2Controller.java

```diff
@@ -63,29 +63,30 @@
   private ServiceModelToSwagger2Mapper mapper;
 
   @Autowired
-  private JsonSerializer jsonSerializer;
+  private MultiFormatSerializer multiFormatSerializer;
 
   @ApiIgnore
   @RequestMapping(value = "${springfox.documentation.swagger.v2.path:" + DEFAULT_URL + "}",
       method = RequestMethod.GET, produces = { APPLICATION_JSON_VALUE, HAL_MEDIA_TYPE })
   public
   @ResponseBody
-  ResponseEntity<Json> getDocumentation(
+  ResponseEntity<RawOutput> getDocumentation(
+      @RequestParam(value = "format", required = false) String format,
```

Comment:
Thats why I suggested a default value. Since many browsers will set the header to `*/*` or equivalent, we can override it to `application/json`.

About the format parameter, Im hesitant to tunnel formats via query string, I'd rather go with a slew of accepted (not a standard) media types for yaml `application/yaml`, `application/yml`, `text/yaml`  etc. [as suggested here](http://stackoverflow.com/questions/332129/yaml-mime-type).